subprojects/version-control/src/main/java/org/gradle/vcs/git/internal/GitVersionControlSystem.java

```diff
@@ -106,15 +126,22 @@ private String getRemoteForUrl(Repository repository, URI url) {
     }
 
     // This is a horrible hack to work around a bug in how jgit stores and expects file urls for remotes.
-    private String safeRepositoryUrl(URI url) {
+    private static String safeRepositoryUrl(URI url) {
         String urlString = url.toString();
         if (urlString.startsWith("file:/") && !urlString.startsWith("file:///")) {
```

Comment:
Cannot use `URI.getScheme()` because it returns `file` for both `file:/example` and `file:///example`. This code specifically needs to convert the former to the later, while leaving the later alone.

I've filed a bug with jgit, but also rearranged the code a bit to make things more readable.